src/LogikUI/File/FileManager.cs

```diff
@@ -0,0 +1,337 @@
+                Output.WriteWarning(args.Message);
+            else
+                throw args.Exception;
+        }
+
+        /// <summary>
+        /// Parses a project file. The results are saved in FileManager.Wires, FileManager.Components, and FileManager.Labels.
+        /// </summary>
+        /// <param name="filename">The path of the project file.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidProjectDataException">The XML-file doesn't match with the schema.</exception>
+        /// <exception cref="ArgumentException">Unable to access the file.</exception>
+        public static void Load(string filename)
+        {
+            Vector2i getPos(XmlNode pos)
+            {
+                int x = int.Parse(pos.Attributes["x"].InnerText);
+                int y = int.Parse(pos.Attributes["y"].InnerText);
+
+                return new Vector2i(x, y);
+            }
+
+            //HACK: What does "Could not find schema information for the element '<element>'." for every element mean!?
+
+            XmlSchemaSet schema = new XmlSchemaSet();
+            // currentDirectory = ~\bin\Debug\netcoreapp3.1
+            string path = Directory.GetCurrentDirectory() + "\\..\\..\\..\\File";
+            schema.Add("http://www.w3.org/2001/XMLSchema", path + "\\structure.xsd");
+
+            XmlDocument doc = new XmlDocument();
+            doc.Schemas = schema;
+            XmlReader? reader;
+
+            XmlReaderSettings settings = new XmlReaderSettings();
+            settings.ValidationType = ValidationType.Schema;
+            settings.ValidationFlags |= XmlSchemaValidationFlags.ProcessInlineSchema;
+            settings.ValidationFlags |= XmlSchemaValidationFlags.ProcessSchemaLocation;
+            settings.ValidationFlags |= XmlSchemaValidationFlags.ReportValidationWarnings;
+            settings.ValidationEventHandler += new ValidationEventHandler(ValidationCallBack);
+
+            try
+            {
+                reader = XmlReader.Create(filename, settings);
+
+                doc.Load(reader);
+
+                #region Parse Wires
+
+                foreach (var _wire in doc.SelectNodes("/circuit/wires"))
+                {
+                    if (_wire != null)
+                    {
+                        XmlNode wire = ((XmlNode)_wire).FirstChild;
+
+                        Vector2i start = getPos(wire.SelectSingleNode("from"));
+                        Vector2i end = getPos(wire.SelectSingleNode("to"));
+
+                        if (start.X == end.X)
+                        {
+                            int length = Math.Abs(Math.Abs(end.Y) - start.Y);
+
+                            Wires.Add(new Wire(start, length, Direction.Horizontal));
+                        }
+                        else if (start.Y == end.Y)
+                        {
+                            int length = Math.Abs(Math.Abs(end.X) - start.X);
```

Comment:
Does this do the right thing?
I think it would be more clear to do `Math.Abs(end.X) - Math.Abs(start.X)`.